inst/include/common/model.hpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * File:   model.hpp
+ *
+ * Author: Matthew Supernaw, Andrea Havron
+ * National Oceanic and Atmospheric Administration
+ * National Marine Fisheries Service
+ * Sustainable Fisheries Division
+ * Email: matthew.supernaw@noaa.gov, andrea.havron@noaa.gov
+ *
+ * Created on September 30, 2021, 1:08 PM
+ *
+ * This File is part of the NOAA, National Marine Fisheries Service
+ * Fisheries Integrated Modeling System project.
+ *
+ * This software is a "United States Government Work" under the terms of the
+ * United States Copyright Act.  It was written as part of the author's official
+ * duties as a United States Government employee and thus cannot be copyrighted.
+ * This software is freely available to the public for use. The National Oceanic
+ * And Atmospheric Administration and the U.S. Government have not placed any
+ * restriction on its use or reproduction.  Although all reasonable efforts have
+ * been taken to ensure the accuracy and reliability of the software and data,
+ * the National Oceanic And Atmospheric Administration and the U.S. Government
+ * do not and cannot warrant the performance or results that may be obtained by
+ * using this  software or data. The National Oceanic And Atmospheric
+ * Administration and the U.S. Government disclaim all warranties, express or
+ * implied, including warranties of performance, merchantability or fitness
+ * for any particular purpose.
+ *
+ * Please cite the author(s) in any work or product based on this material.
+ *
+ */
+#ifndef FIMS_COMMON_MODEL_HPP
+#define FIMS_COMMON_MODEL_HPP
+
+#include <memory>
+
+#include "../population_dynamics/population/population.hpp"
+#include "../population_dynamics/fleet/fleet.hpp"
+
+
+namespace fims{
+
+    /**
+     * Model base
+     */
+    template<typename T>
+    class Model{
```

Comment:
Need a better class comment here - preferably with example https://google.github.io/styleguide/cppguide.html#Class_Comments

Reply:
@msupernaw, can you add class comment as you are more familiar its functionality?